Bottom: fab6fe84ac0a6c40f876fe7dd1b5d6c4eebda580
Top:    d24c22bc8c6bb8268717e15c654972f5fe05c929
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-17 19:53:59 +0000

polygon1.c: Fix unitialised output polygon upon when hitting allocation failure

This was a bug present in the poly_Boolean() routine. We need to set the output
polygon to NULL if we are going to return early on memory failure. This was
noticed during general review after observing problems due to our copy routines
incorrectly returning FALSE when copying a NULL polygon.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 0498d36..a8d9391 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2327,6 +2327,8 @@ poly_Boolean (const POLYAREA * a_org, const POLYAREA * b_org,
 {
   POLYAREA *a = NULL, *b = NULL;
 
+  *res = NULL;
+
   if (!poly_M_Copy0 (&a, a_org) || !poly_M_Copy0 (&b, b_org))
     return err_no_memory;
